pydicom/filereader.py

```diff
@@ -115,15 +116,10 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
-    tag_set = set()
-    if specific_tags is not None:
-        for tag in specific_tags:
-            if isinstance(tag, str):
-                tag = Tag(tag_for_keyword(tag))
-            if isinstance(tag, BaseTag):
-                tag_set.add(tag)
-        tag_set.add(Tag(0x08, 0x05))
-    has_tag_set = len(tag_set) > 0
+    tag_set = {Tag(tag) for tag in specific_tags} if specific_tags else []
+    has_tag_set = bool(tag_set)
+    if has_tag_set:
```

Comment:
Why not use `if tag_set`? Is the type checker complaining here? It shouldn't, I think...

Reply:
No, but to keep the refactor minor I kept to the 'pattern' of the old code. Since I had `has_tag_set` already, I figured I'd use it